bigquery_etl/cli/query.py

```diff
@@ -1479,64 +1479,77 @@ def deploy(
             existing_schema.to_json_file(Path(tmp_schema_file.name))
             bigquery_schema = client.schema_from_json(tmp_schema_file.name)
 
-            try:
-                # destination table already exists, update schema
-                table = client.get_table(full_table_id)
-                table.schema = bigquery_schema
-                client.update_table(table, ["schema"])
-                click.echo(f"Schema updated for {full_table_id}")
-            except NotFound:
-                # no destination table, create new table based on schema and metadata
-                metadata = Metadata.of_query_file(query_file_path)
-                new_table = bigquery.Table(full_table_id, schema=bigquery_schema)
-
-                if metadata.bigquery and metadata.bigquery.time_partitioning:
-                    new_table.time_partitioning = bigquery.TimePartitioning(
-                        metadata.bigquery.time_partitioning.type.bigquery_type,
-                        field=metadata.bigquery.time_partitioning.field,
-                        require_partition_filter=(
-                            metadata.bigquery.time_partitioning.require_partition_filter
-                        ),
-                        expiration_ms=metadata.bigquery.time_partitioning.expiration_ms,
-                    )
+        try:
+            table = client.get_table(full_table_id)
+        except NotFound:
+            table = bigquery.Table(full_table_id)
 
-                if metadata.bigquery and metadata.bigquery.clustering:
-                    new_table.clustering_fields = metadata.bigquery.clustering.fields
+        table.schema = bigquery_schema
 
-                client.create_table(new_table)
-                click.echo(f"Destination table {full_table_id} created.")
+        if table.created is None:
+            _attach_metadata(query_file_path, table)
+            client.create_table(table)
+            click.echo(f"Destination table {full_table_id} created.")
         else:
-            click.echo(f"No schema file found for {query_file}")
+            client.update_table(table, ["schema"])
+            click.echo(f"Schema updated for {full_table_id}.")
+
+
+def _attach_metadata(query_file_path: Path, table: bigquery.Table) -> None:
+    metadata = Metadata.of_query_file(query_file_path)
+
+    table.description = metadata.description
+    table.friendly_name = metadata.friendly_name
+
+    if metadata.bigquery and metadata.bigquery.time_partitioning:
+        table.time_partitioning = bigquery.TimePartitioning(
+            metadata.bigquery.time_partitioning.type.bigquery_type,
+            field=metadata.bigquery.time_partitioning.field,
+            require_partition_filter=(
+                metadata.bigquery.time_partitioning.require_partition_filter
+            ),
+            expiration_ms=metadata.bigquery.time_partitioning.expiration_ms,
+        )
+
+    if metadata.bigquery and metadata.bigquery.clustering:
```

Comment:
Is the `metadata.bigquery` necessary here?

Reply:
Yup, it's optional: https://github.com/mozilla/bigquery-etl/blob/f6b20988b5f19c9ce5ce04eda405a7bae89e6ff7/bigquery_etl/metadata/parse_metadata.py#L131